opm/material/components/CO2Tables.hpp

```diff
@@ -57,16 +57,51 @@ struct co2TabulatedEnthalpyTraits {
     static const Scalar vals[200][500];
 };
 
+template<class Scalar = double, class ContainerT = std::vector<double>>
 class CO2Tables
 {
 public:
-    Opm::UniformTabulated2DFunction<double> tabulatedEnthalpy;
-    Opm::UniformTabulated2DFunction<double> tabulatedDensity;
+    Opm::UniformTabulated2DFunction<Scalar, ContainerT> tabulatedEnthalpy;
+    Opm::UniformTabulated2DFunction<Scalar, ContainerT> tabulatedDensity;
     static constexpr double brineSalinity = 1.000000000000000e-01;
 
     CO2Tables();
+
+    CO2Tables(const Opm::UniformTabulated2DFunction<Scalar, ContainerT>& enthalpy,
+              const Opm::UniformTabulated2DFunction<Scalar, ContainerT>& density)
+        : tabulatedEnthalpy(enthalpy), tabulatedDensity(density)
+    {
+    }
+
+    const Opm::UniformTabulated2DFunction<Scalar, ContainerT>& getTabulatedEnthalpy() const {
+        return tabulatedEnthalpy;
+    }
+
+    const Opm::UniformTabulated2DFunction<Scalar, ContainerT>& getTabulatedDensity() const {
+        return tabulatedDensity;
+    }
 };
 
 } // namespace Opm
 
+namespace Opm::gpuistl {
+    template <class Scalar, class ContainerType, class ViewType>
+    CO2Tables<Scalar, ViewType>
+    make_view(const CO2Tables<Scalar, ContainerType>& oldCO2Tables) {
+        Opm::UniformTabulated2DFunction<double, ViewType> newEnthalpy = make_view<Scalar, ContainerType, ViewType>(oldCO2Tables.getTabulatedEnthalpy());
+        Opm::UniformTabulated2DFunction<double, ViewType> newDensity = make_view<Scalar, ContainerType, ViewType>(oldCO2Tables.getTabulatedDensity());
+
+        return CO2Tables<Scalar, ViewType>(newEnthalpy, newDensity);
```

Comment:
Should perhaps std::move() these objects? I must admit I am not 100% sure how this would run (copying too many times or not) when involving the GPU here. And if a view is always light-weight then maybe it does not matter.

Reply:
making and copying views happens in constant time, only the pointers are handled